extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/KinesisRegion.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.indexing.kinesis;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import org.apache.druid.java.util.common.IAE;
+import org.apache.druid.java.util.common.StringUtils;
+
+import java.util.EnumSet;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public enum KinesisRegion
+{
+  US_EAST_2,
+  US_EAST_1,
+  US_WEST_1,
+  US_WEST_2,
+  AP_NORTHEAST_1,
+  AP_NORTHEAST_2,
+  AP_NORTHEAST_3,
+  AP_SOUTH_1,
+  AP_SOUTHEAST_1,
+  AP_SOUTHEAST_2,
+  CA_CENTRAL_1,
+  CN_NORTH_1,
+  CN_NORTHWEST_1,
+  EU_CENTRAL_1,
+  EU_WEST_1,
+  EU_WEST_2,
+  EU_WEST_3,
+  SA_EAST_1,
+  US_GOV_WEST_1;
+
+  @JsonCreator
+  public static KinesisRegion fromString(String value)
+  {
+    return EnumSet.allOf(KinesisRegion.class)
+                  .stream()
+                  .filter(x -> x.toString().equals(value))
+                  .findFirst()
+                  .orElseThrow(() -> new IAE("Region must be one of: %s", getNames()));
+  }
+
+  private static List<String> getNames()
+  {
+    return EnumSet.allOf(KinesisRegion.class).stream().map(KinesisRegion::toString).collect(Collectors.toList());
+  }
+
+  public String getEndpoint()
+  {
+    return StringUtils.format("kinesis.%s.amazonaws.com%s", toString(), toString().startsWith("cn-") ? ".cn" : "");
+  }
+
+  @Override
+  @JsonValue
+  public String toString()
+  {
+    return StringUtils.toLowerCase(name()).replace('_', '-');
```

Comment:
It looks like the json serialized KinesisRegion will have the different name from the enum which can make deserialization fail.

I would suggest to add a new method instead of overriding `toString`. Also please add a unit test for json serde of this class.

Reply:
Thanks, but I still think it's better to not override `toString` since we expect `toString` returns its name for enum types. What do you think?

Reply:
removed toString() and added serialize()

Reply:
added